bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/LspUtils.java

```diff
@@ -46,5 +53,82 @@ private static Shell getActiveShell() {
 			return PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell();
 		return null;
 	}
+	
+	public static boolean isFileOpenedInLspEditor(URI uri) {
+		if (uri == null) {
+			return false;
+		}
+		var activeEditors = getActiveEditors();
+		if (activeEditors != null && activeEditors.length > 0) {
+			for (IEditorReference editor : activeEditors) {
+				IEditorInput editorInput = null;
+				URI editorUnputURI = null;
+				try {
+					editorInput = editor.getEditorInput();
+				} catch (PartInitException e) {
+					LspPlugin.logError(e.getMessage(), e);
+					continue;
+				}
+				
+				if (editorInput instanceof IURIEditorInput) {
+					editorUnputURI = ((IURIEditorInput) editorInput).getURI();
+				} else if (editorInput instanceof FileEditorInput) {
+					editorUnputURI = ((FileEditorInput) editorInput).getFile().getLocationURI();
+				}
+
+				if (uri.equals(editorUnputURI)) {
+					// should return false when an external header file with same URI is opened in a LSP editor 
+					// and non LSP editor and tab switching from a non LSP editor to the tab with the file in the non LSP editor:
+					return LspPlugin.LSP_C_EDITOR_ID.equals(editor.getEditor(true).getEditorSite().getId()) && isLspEditorActive();
+				}
+			}
+			// the file has not been opened yet -> goto definition/declaration case
+			return isLspEditorActive();
+		}
+		return false;
+	}
+	
+	public static boolean isFileOpenedInLspEditor(IEditorInput editorInput) {
+		if (editorInput == null) {
+			return false;
+		}
+		var activeEditors = getActiveEditors();
+		if (activeEditors != null && activeEditors.length > 0) {
+			for (IEditorReference editor : activeEditors) {
+				IEditorInput editorInputFromEditor = null;
+				try {
+					editorInputFromEditor = editor.getEditorInput();
+				} catch (PartInitException e) {
+					LspPlugin.logError(e.getMessage(), e);
+					continue;
+				}
+				if (editorInput.equals(editorInputFromEditor)) {
+					return LspPlugin.LSP_C_EDITOR_ID.equals(editor.getEditor(true).getEditorSite().getId());
+				}
+			}
+			// the file has not been opened yet:
+			return isLspEditorActive();
+		}
+		return false;
+	}
+	
+	public static IEditorReference[] getActiveEditors() {
```

Comment:
Perhaps this method is not needed, see other comments from its call sites

Reply:
see comment above https://github.com/eclipse-cdt/cdt-lsp/pull/61#discussion_r1191360013